radiography/src/main/java/radiography/ViewFilter.kt

```diff
@@ -9,18 +10,28 @@ public interface ViewFilter {
    * @return true to keep the view in the output of [Radiography.scan], false to filter it out.
    */
   public fun matches(view: Any): Boolean
+}
+
+/**
+ * Base class for implementations of [ViewFilter] that only want to filter instances of a specific
+ * type. Instances of other types are always "matched" by this filter.
+ */
+internal abstract class TypedViewFilter<in T : Any>(
```

Comment:
Made both these abstract type-specific classes `internal` for now. If it turns out we actually need them to be more open, we can always make this public in a minor release.